src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -218,6 +218,11 @@ private Run CreateSarifRun()
                     _originalUriBasePath = "file:/" + _originalUriBasePath;
                 }
 
+                if (_originalUriBasePath.EndsWith(":"))
+                {
+                    _originalUriBasePath = _originalUriBasePath + "/";
```

Comment:
>"/" [](start = 66, length = 3)

You know, the spec says (§3.14.14, originalUriBaseIds property):

> The values of the `uri` properties in the `artifactLocation` objects in `originalUriBaseIds`:
>   • SHALL end with a single forward slash.

It's true for _all_ the `uri` values, not just the ones that happen to look like `C:`. So replace these three lines with:
```
if (!originalUriBasePath.EndsWith("/")) { _originalUriBasePath += "/"; }
```

Once you've merged this PR, consider adding a utility method like `SarifUtilities.EnsureTrailingSlash`, and make sure this logic happens _everywhere_ we create a `run.originalUriBaseIds`. Or just file an issue :-)

Reply:
thanks, updated. and i extracted the method to bring under unit test, this includes linux platform related change as well, so i added that as another test case.

---
In reply to: [297943211](https://github.com/microsoft/sarif-sdk/pull/1547#discussion_r297943211) [](ancestors = 297943211)

Reply:
oh! i just noticed the second part of your suggestion reg moving this to a utility, thats a great idea! will add an issue.

---
In reply to: [297959359](https://github.com/microsoft/sarif-sdk/pull/1547#discussion_r297959359) [](ancestors = 297959359,297943211)